File: app/src/main/java/zechs/zplex/utils/ConverterUtils.java
package zechs.zplex.utils;

import android.annotation.SuppressLint;

public class ConverterUtils {

    @SuppressLint("DefaultLocale")
    public static String getSize(long size) {
        String s;
        double kb = Double.parseDouble(String.valueOf(size)) / 1024;
        double mb = kb / 1024;
        double gb = mb / 1024;
        double tb = gb / 1024;
        if (size < 1024L) {
            s = size + " Bytes";
        } else if (size < 1024L * 1024) {
            s = String.format("%.2f", kb) + " KB";
        } else if (size < 1024L * 1024 * 1024) {
            s = String.format("%.2f", mb) + " MB";
        } else if (size < 1024L * 1024 * 1024 * 1024) {
            s = String.format("%.2f", gb) + " GB";
        } else {
            s = String.format("%.2f", tb) + " TB";
        }
        return s;
    }
}

File: app/src/main/kotlin/zechs/zplex/utils/ConverterUtils.kt
package zechs.zplex.utils

class ConverterUtils {
    companion object {
        fun getSize(size: Long): String {
            val s: String
            val kb = size.toString().toDouble() / 1024
            val mb = kb / 1024
            val gb = mb / 1024
            val tb = gb / 1024
            s = when {
                size < 1024L -> "$size Bytes"
                size < 1024L * 1024 -> String.format("%.2f", kb) + " KB"
                size < 1024L * 1024 * 1024 -> String.format("%.2f", mb) + " MB"
                size < 1024L * 1024 * 1024 * 1024 -> String.format("%.2f", gb) + " GB"
                else -> String.format("%.2f", tb) + " TB"
            }
            return s
        }
    }
}